resources/resources.go

```diff
@@ -61,13 +61,40 @@ func (rp ResourcePlan) ApplyPlan(name string, vc VaultClient) (Resource, error)
 		ScaleOutCooldown:    ucd,
 		NomadToComputeRatio: rp.NomadToComputeRatio,
 		Name:                name,
+		nextResetTime:       time.Unix(1<<63-62135596801, 999999999),
 	}, nil
 }
 
+func fmtDuration(d time.Duration) string {
+	d = d.Round(time.Minute)
+	h := d / time.Hour
+	d -= h * time.Hour
+	m := d / time.Minute
+	return fmt.Sprintf("%02d:%02d", h, m)
+}
+
 // Scale receives a desired nomad count and scales both nomad + ec2 accordingly
 func (res *EC2NomadResource) Scale(desiredNomadCount int, vc *VaultClient) error {
-	// check if its a scale out or scale in
+	now := time.Now()
+	maxTime := time.Unix(1<<63-62135596801, 999999999)
 
+	if now.After(res.nextResetTime) {
+		if err := res.NomadClient.RestartNomadAlloc(); err != nil {
+			resetDuration := res.getResetDuration()
+			res.nextResetTime = now.Add(resetDuration)
+			logging.Info("[restart log] cancel allocation failure. starting timer to retry cancel allocation in %v", resetDuration)
+		} else {
+
+			res.nextResetTime = maxTime
+			logging.Info("[restart log] cancel allocation success")
+		}
+	}
+
+	if time.Since(res.lastScaledTime) < res.ScaleInCooldown {
+		return fmt.Errorf("Too soon to scale again")
```

Comment:
Nit: You might want to capitalize first char more consistently for messages?

Reply:
👍